### PR TITLE
support VSCode Insiders build alongside Stable

### LINKED ISSUE #3011: Visual Studio Code Insiders is not recognised as en editor
I recently updated to the insiders version of Visual Studio Code, but now Github Desktop says that there are no editors installed.

### Version
**GitHub Desktop version:** [1.0.3]

**OS version:** [Windows 10]

### Steps to Reproduce

1. [Install Desktop and VSCode insiders]
1. [Try to open a project in the editor]

**Expected behavior:** [Open the project in VSCode]

**Actual behavior:** [Gives an error about no editor]
**Reproduces how often:** [100%]

### PR DESCRIPTION
**Pull request by say25**
- All Platforms

Addresses #3011 @shiftkey's comment 

> @MSathieu maybe? I don't think we can have multiple enum values mapped to a single string (so we can separate the logic for ExternalEditor.VisualStudioCodeInsiders but make it appear in the UI as 'Visual Studio Code') so maybe we'll need to call it 'Visual Studio Code (Insiders)'.

Its pretty typical for those with VS Code Insiders installed to also have VS Code (vanilla) installed side by side. This allows each to be selected.

### CODE REVIEW

**Comment by shiftkey** [comment]
@say25 I think we stumbled upon the linting issue you're facing in #3449 - try merging `master` into this branch and see if `yarn lint:fix` works again...

**Comment by say25** [comment]
will do tonight

**Comment by say25** [comment]
@shiftkey It seems that running `yarn eslint:fix` locally doesn't do make any further changes to the failing files yet still errors on the build agents :/

**Comment by shiftkey** [comment]
@say25 you might have to refresh your packages by running `yarn` again before trying to run `yarn lint` or `yarn lint:fix`.

**Comment by say25** [comment]
Seems like you were right @shiftkey I'm a yarn noob

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -83,6 +92,7 @@ function getExecutableShim(
     case ExternalEditor.Atom:
       return Path.join(installLocation, 'bin', 'atom.cmd')
     case ExternalEditor.VisualStudioCode:
+    case ExternalEditor.VisualStudioCodeInsiders:
```
[the remark this answers is not in the document]
Comment (say25) [reply app/src/lib/editors/win32.ts]: Weird but fixed

**Comment by shiftkey** [comment]
@say25 just need to fix the conflict from merging #3467 and this should be good to merge.

**Comment by shiftkey** [comment]
I've just pushed a lint fix for that. I'll drop myself off the review and give someone else a chance to give this a final 👍 .

**Comment by say25** [comment]
@shiftkey thanks. Dont have yarn install on my windows box :/ probably should do that now.

**Comment by shiftkey** [comment]
@say25 yeah, we've switched over to using it for lockfile support (NPM@5 has some limitations that affect cross-platform libraries that remain a blocker)

**Pull request merged**